File: trail2/함수를 이용한 합과 소수 판별/use-functions-to-determine-sums-and-decimals.cpp
#include <iostream>
using namespace std;

bool isEvenSum(int num) {
    int sum = 0;

    while (num != 0) {
        sum += num % 10;
        num /= 10;
    }

    return sum % 2 == 0;
}

bool isPrime(int num) {
    for (int i = 2; i < num; i++) {
        if (num % i == 0) 
            return false;
    }

    return true;
}

int main() {
    // Please write your code here.
    int a, b, cnt = 0;
    cin >> a >> b;

    for (int num = a; num <= b; num++) {
        if (isPrime(num) && isEvenSum(num)) cnt++;
    }

    cout << cnt << endl;

    return 0;
}

File: trail2/함수를 이용한 합과 소수 판별/use-functions-to-determine-sums-and-decimals.py
a, b = tuple(map(int, input().split()))

def is_sum_even(num):
    sum = 0
    while num > 0:
        sum += num % 10
        num //= 10
    
    if sum % 2 == 0:
        return True
    else:
        return False

def is_prime(num):
    if num == 1:
        return False

    for i in range(2, num):
        if (num % i == 0):
            return False
    return True

cnt = 0
for i in range(a, b+1):
    if (is_prime(i) and is_sum_even(i)):
        cnt += 1

print(cnt)